enhancements/kube-apiserver/api-webhook-supportability.md

```diff
@@ -0,0 +1,506 @@
+
+The kube-apiserver operator will go degraded
+
+- if webhooks for virtual resources (SubjectAccessReview, ...) exist
+- if webhooks's service does not exist, or has no endpoints.
+
+Virtual resources are those that are not persisted in etcd (labelled as "virtual" in the
+[Critical OCP API Resource spreadsheet](https://docs.google.com/spreadsheets/d/18A-bGQTcXrCB_sHAHBtzmsz0ma9fvFBtt5PY-VZ8T-Y))):
+
+- bindings
+- `*`reviews.
+
+We are considering to make kube-controller-manager operator go degraded in parallel because
+garbage collection and other dynamic controllers will stop working when not all resources are
+readable because of missing CRD conversion webhook services. This happened many times in
+customer clusters in the past, and today can only be diagnosed by looking into kube-apiserver logs.
+But connecting to dots between broken garbage collection and a missing service is hard.
+
+Note: OLM today does intentionally not delete CRDs on operator uninstall. With this enhancement in place
+every cluster that has a dangling webhooks service reference will degrade. Hence, a `olm uninstall <operator-with-CRD-conversion>`
+will break the cluster. Also note that the cluster is already broken in that situation. So to degrade
+is the correct behaviour, not a regression.
+
+### Informational Alerts
+
+- webhook installed for core resources that are not whitelisted (via soft approval; restricted to
+  openshift-`*` namespaces)
+- webhook installed for security-sensitive resources  (TokenReviews, oauth tokens+clients) that are
+  not whitelisted (including non-openshift-`*` components).
+
+  This alert is not expressible with the metrics we have. We have to find a way either via
+  artificial metrics from the operator or by moving to some condition instead.
+
+Core resources are those shipped in a stock OCP cluster.
+
+Security sensitive resources (labelled with "security" in the [Critical OCP API Resource
+spreadsheet](https://docs.google.com/spreadsheets/d/18A-bGQTcXrCB_sHAHBtzmsz0ma9fvFBtt5PY-VZ8T-Y)):
+
+- secrets
+- serviceaccounts
+- mutatingwebhookconfigurations
+- validatingwebhookconfigurations
+- tokenreviews
+- certificatesigningrequests
+- credentialsrequests
+- oauthaccesstokens
+- oauthauthorizetokens
+- oauthclientauthorizations
+- oauthclients
+- useroauthaccesstokens
+- routes.
+
+### "Soft" Approval Process
+
+For RH-internal add-ons (those whose service lives in `openshift-*` namespaces) we add a
+```yaml
+webhook-approved.openshift.io: https://github.com/openshift/enhancement/pull/<pr>
+```
+annotation on CRDs with webhook conversion and admission registrations objects (similarly to
+`api-approved.openshift.io`). This will work as a white-list marker to hook into core
+resources. These webhooks are excluded from the informational alerts for core resources of the
+previous section. No other action is taken beyond the informational alert i.e. to stop those
+webhooks to work.
+
+### Extension of OpenShift Enhancement Template
```

Comment:
@sttts I don't see a PR nor the template updated, yet

Reply:
Yep, will follow-up.